packages/provider/src/transports/unreal-transport/unreal-message-provider.ts

```diff
@@ -0,0 +1,121 @@
+import { OpenWalletIntent, ProviderMessage, InitState, WindowSessionParams } from '../../types'
+import { BaseProviderTransport } from '../base-provider-transport'
+import { base64EncodeObject } from '@0xsequence/utils'
+import { overrideLogs } from './overridelogs'
+
+let registeredUnrealMessageProvider: UnrealMessageProvider | undefined
+
+// all lowercase is an annoying limitation of Unreal CEF BindUObject
+interface UnrealInjectedSequenceJSWindow {
+  ue?: {
+    sequencewallettransport?: {
+      onmessagefromwallet?: (message: ProviderMessage<any>) => void
+      sendmessagetowallet: (message: string) => void
+    }
+  }
+}
+
+declare const window: Window & typeof globalThis & UnrealInjectedSequenceJSWindow
+
+/**
+ * Initialized on dApp side
+ */
+export class UnrealMessageProvider extends BaseProviderTransport {
+  private walletURL: URL
+
+  constructor(walletAppURL: string) {
+    super()
+    this.walletURL = new URL(walletAppURL)
+  }
+
+  register = () => {
+    overrideLogs('dapp')
+    if (registeredUnrealMessageProvider) {
+      // overriding the registered message provider
+      registeredUnrealMessageProvider.unregister()
+      registeredUnrealMessageProvider = this
+    }
+
+    // listen for incoming messages from wallet
+    if (window.ue?.sequencewallettransport) {
+      window.ue.sequencewallettransport.onmessagefromwallet = this.onUnrealCallback
+    }
+    registeredUnrealMessageProvider = this
+
+    this._registered = true
+    console.log('registering transport!')
+  }
+
+  unregister = () => {
+    this._registered = false
+    this.closeWallet()
+
+    // disable message listener
+    if (registeredUnrealMessageProvider === this) {
+      registeredUnrealMessageProvider = undefined
+    }
+    if (window.ue?.sequencewallettransport?.onmessagefromwallet === this.onUnrealCallback) {
+      delete window.ue.sequencewallettransport.onmessagefromwallet
+    }
+
+    // clear event listeners
+    this.events.removeAllListeners()
+  }
+
+  openWallet = (path?: string, intent?: OpenWalletIntent, networkId?: string | number): void => {
+    if (this.isOpened()) {
+      // TODO focus wallet
+      console.log('wallet already open!')
+      return
+    }
+
+    console.log('opening wallet!')
+    // Instantiate new walletURL for this call
+    const walletURL = new URL(this.walletURL.href)
+    const windowSessionParams = new WindowSessionParams()
+
+    if (path && path !== '') {
+      walletURL.pathname = path.toLowerCase()
```

Comment:
Is it correct to always lower case the path? Shouldn't we use the path as provided by the dapp?